File: native-android/app/src/main/java/com/iganapolsky/randomtimer/monetization/ProSoundAccess.kt
package com.iganapolsky.randomtimer.monetization

import com.iganapolsky.randomtimer.domain.model.SoundType

/** Free-tier Pro sound access via subscription or rewarded-ad trial unlock. */
object ProSoundAccess {
    fun canEquipProSound(
        isPro: Boolean,
        hasTrialUnlock: Boolean,
    ): Boolean = isPro || hasTrialUnlock

    fun shouldConsumeTrialOnEquip(
        isPro: Boolean,
        hasTrialUnlock: Boolean,
        previousSound: SoundType,
        newSound: SoundType,
    ): Boolean =
        !isPro &&
            hasTrialUnlock &&
            newSound in SoundType.PRO &&
            previousSound !in SoundType.PRO
}

File: native-ios/RandomTimer/Sources/Services/ProSoundAccess.swift
import Foundation

/// Free-tier Pro sound access via subscription or rewarded-ad trial unlock.
enum ProSoundAccess {
    static func canEquipProSound(isPro: Bool, hasTrialUnlock: Bool) -> Bool {
        isPro || hasTrialUnlock
    }

    static func shouldConsumeTrialOnEquip(
        isPro: Bool,
        hasTrialUnlock: Bool,
        previousSound: SoundType,
        newSound: SoundType
    ) -> Bool {
        !isPro &&
            hasTrialUnlock &&
            newSound.isPro &&
            !previousSound.isPro
    }
}
